### DISCUSSION TITLE
API Rule only if relation exists between two collections

### DISCUSSION THREAD

**Discussion by nattensleplubo** [pocketbase/pocketbase#4556] [Q&A] [answered]
Hello, basically my users collection have a one to many relation with another collection named 'students'. And right now the current API rule is set to : id = @request.auth.id. But that means anyone logged in can delete a student. I want it so that only the user parent of a student can delete the student. What would be the rule for this, the doc isn't clear about this and i can't find any answer about it. Thank you very much!

**Comment by ganigeorgiev** [comment]
> And right now the current API rule is set to : id = @request.auth.id. But that means anyone logged in can delete a student. 

If you set the above API rule for the `students` collection it means that only the authenticated user that has id matching your student id can delete the students record, which I guess it is not what you are looking for.

I'm not sure that I fully understand your collections structure from your description but if we assume that the multiple `relation` field is named `students`, then one possible delete API rule for the `students` collection could be:

```js
// the request must be authenticated
@request.auth.id != "" &&
// the authenticated user must have a relation to the current deleting student
// (roughly translates to: "at least one of the user.students ids must be equal to the current student id")
@request.auth.students.id ?= id
```

If the above doesn't help, please be more specific about your collections structure and the available fields (your can also export your collections from the _Admin UI > Settings > Export collections_).

**Reply by nattensleplubo** [reply]
![image](https://github.com/pocketbase/pocketbase/assets/86134104/4831cb80-7967-445b-b680-1ae7dfe0326c)
This is my users collection, this one has two students (named eleves here), i want only this user to be able to delete / modify them. Is that what you said ?

**Reply by ganigeorgiev** [reply, marked as answer]
Yes, the above rule should be correct if you replace `@requst.auth.students` with `@request.auth.eleves`:

```js

@request.auth.id != "" &&
@request.auth.eleves.id ?= id
```

**Reply by nattensleplubo** [reply]
Ok thank you very much !! Just started learning web so kinda new to this stuff
